src/azul/azulclient.py

```diff
@@ -381,36 +381,39 @@ def delete_bundle(self, catalog: CatalogName, bundle_uuid, bundle_version):
 
     def deindex(self, catalog: CatalogName, sources: Iterable[str]):
         plugin = self.repository_plugin(catalog)
-        sources = [plugin.resolve_source(s).id for s in sources]
+        sources = [plugin.resolve_source(s) for s in sources]
+        source_ids = [s.id for s in sources]
         es_client = ESClientFactory.get()
         indices = ','.join(self.index_service.index_names(catalog))
         query = {
-            "query": {
-                "bool": {
-                    "should": [
+            'query': {
+                'bool': {
+                    'should': [
                         {
-                            "terms": {
-                                "sources.id.keyword": sources
+                            'terms': {
+                                # Aggregate documents
+                                'sources.id.keyword': source_ids
                             }
                         },
                         {
-                            "terms": {
-                                "source.id.keyword": sources
+                            'terms': {
+                                # Contribution documents
+                                'source.id.keyword': source_ids
                             }
                         }
                     ]
                 }
             }
         }
-        logger.info('Deindexing sources: %s from catalog: %s', sources, catalog)
+        logger.info('Deindexing sources %r from catalog %r', sources, catalog)
         logger.debug('Using query: %r', query)
         response = es_client.delete_by_query(index=indices, body=query, slices='auto')
-        if len(response['failures']) > 0:
-            if response['version_conflicts'] > 0:
+        if response['failures']:
+            if response['version_conflicts']:
```

Comment:
I didn't actually ask for this one. I prefer the numeric comparison here because of `-1` or when the API changes this from number a dictionary with more detail.

On that note I think I actually prefer what you had originally for both comparisons. Sorry for wavering.